common/tar.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef MENDER_COMMON_TAR_HPP
+#define MENDER_COMMON_TAR_HPP
+
+#include <string>
+#include <vector>
+
+#include <config.h>
+
+#include <common/io.hpp>
+#include <common/expected.hpp>
+
+#ifdef MENDER_TAR_LIBARCHIVE
+#include <linux/archiver.hpp>
+#endif
+
+namespace tar {
+
+namespace expected = mender::common::expected;
+
+using namespace std;
+
+
+struct Buffer {
+	const char *buffer;
+	size_t size;
```

Comment:
Too bad we can't use `std::span` here, because it could give us range checking. But it's C++20, which is too new even for Unix I think.